docs/content/docs/try-flink/write_flink_program_with_sql.md

```diff
@@ -0,0 +1,269 @@
+
+Execute the following query in the SQL client:
+
+```sql
+CREATE TEMPORARY TABLE twitch_stream (
+  `user_name` VARCHAR(2147483647),
+  `game_name` VARCHAR(2147483647),
+  `viewer_count` INT,
+  `started_at` TIMESTAMP_LTZ(3),
+  `location` VARCHAR(2147483647),
+  proctime AS PROCTIME()
+) WITH (
+  'connector' = 'faker',
+  'fields.user_name.expression' = '#{name.fullName}',
+  'fields.game_name.expression' = '#{book.title}',
+  'fields.viewer_count.expression' = '#{number.numberBetween ''0'',''100''}',
+  'fields.started_at.expression' = '#{date.past ''15'',''SECONDS''}',
+  'fields.location.expression' = '#{country.name}'
+);
+```
+
+You should hopefully see `[INFO] Execute statement succeed.`
+
+Note that a temporary table does not have a catalog configured and will just be available in this session. 
+
+Let's delve a bit into the notion of time. Within data processing systems, there are typically two types of time to reason about: event time (the time at which events actually occurred) and processing time (the time at whch events are observed in the system). The latter is the [simplest notion of time]({{< ref "docs/dev/table/concepts/time_attributes#processing-time" >}}) and is what you will be using via `PROCTIME()`.
+
+
+## Step 4: Run your first continuous SQL query and learn about dynamic tables
+
+Now use the DQL syntax `SELECT` to view the streaming data in this table:
+
+```sql
+SELECT * FROM twitch_stream;
+```
+
+You should see, on the console of the SQL client, a stream of data populating each of the defined fields of the twitch_stream table.  Notice that this table contains a stream of data and is what is known as a [dynamic table]({{< ref "docs/dev/table/concepts/dynamic_tables" >}}).  
+
+Dynamic tables are the fundamental concept behind Flink's SQL support for streaming data. While SQL makes it seem like you are querying a database (or static tables that represent batch data, the tables in Flink are actually dynamic tables that are defined by queries. So instead of running several different queries on the same set of data, you are continuously running one query on a dataset that keeps changing. 
+
+Under the hood, the SQL client submits queries to Flink's JobManager, which works with the TaskManager(s) to assign and monitor query tasks. Have a look at [Flink's architecture]({{< ref "docs/concepts/flink-architecture" >}}) for more detail. 
+
+
+## Step 5: Filter the data
+
+Now try to perform a filter operation on this data stream to specify a subset of the data by using the `WHERE` keyword.  
+
+To find out what gaming streams started in the last 15 minutes, try this query:
+
+```sql
+SELECT game_name, 
+       started_at,
+       proctime, 
+       TIMESTAMPDIFF(MINUTE, started_at, proctime)
+FROM twitch_stream
+WHERE TIMESTAMPDIFF(MINUTE, started_at, proctime) < 15;
+```
+
+You should now see a new table with new datastream results.  
+
+
```

Comment:
top-n isn't really another example of filtering. Maybe insert a new Step 6 (and increment all of the later steps)?